syntax = "proto3";

import "error.proto";

package neptune;

service Discovery {
        rpc Keepalive (stream KeepaliveRequest) returns (stream Servers) {}
	rpc Register (Server) returns (RegisterResponse) {}
	rpc Echo (EchoMsg) returns (EchoMsg) {}
}

message EchoMsg {
  string Msg = 1;
}

message RegisterResponse {
  neptune.CommonError Error = 1;
  int32 Keepalive = 2;
}

message KeepaliveRequest {
  int32 Id = 1;
}

message Server {
  string Type = 1;
  int32 Id = 2;
  string Address = 3;
  repeated string Services = 4;
}

message Servers {
  neptune.CommonError Error = 1;
  repeated Server Servers = 2;
}
